homeassistant/components/forecast_solar/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The Forecast Solar integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from forecast_solar import ForecastSolar
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_AZIMUTH, CONF_DECLINATION, CONF_MODULES_POWER, DOMAIN
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Forecast Solar from a config entry."""
+
+    forecast = ForecastSolar(
+        latitude=entry.data[CONF_LATITUDE],
+        longitude=entry.data[CONF_LONGITUDE],
+        declination=entry.data[CONF_DECLINATION],
+        azimuth=(entry.data[CONF_AZIMUTH] - 180),
+        kwp=(entry.data[CONF_MODULES_POWER] / 1000),
+        damping=0,
+    )
+
+    # Create Forecast Solar instance for this entry
+    coordinator: DataUpdateCoordinator = DataUpdateCoordinator(
+        hass,
+        logging.getLogger(__name__),
+        name=DOMAIN,
+        update_method=forecast.estimate,
+        update_interval=timedelta(minutes=15),
```

Comment:
Data resolution: 1 hour for the free version, so maybe that should be the default?

Reply:
integration is still WIP, the interval will be adjusted

Reply:
Resolved.